Assets/Fungus/Scripts/EventHandlers/DragCompleted.cs

```diff
@@ -123,28 +174,53 @@ public virtual void OnDragExited(Draggable2D draggableObject, Collider2D targetO
         /// </summary>
         public virtual void OnDragCompleted(Draggable2D draggableObject)
         {
-            if (draggableObject == this.draggableObject &&
+            if (this.draggableObjects.Contains(draggableObject) &&
                 overTarget)
             {
                 // Assume that the player will have to do perform another drag and drop operation
                 // to complete the drag again. This is necessary because we don't get an OnDragExited if the
                 // draggable object is set to be inactive.
+                if(draggableRef!=null)
+                {
+                    draggableRef.Value = draggableObject.gameObject;
+                }
+                if(targetRef!=null)
+                {
+                    targetRef.Value = targetObject.gameObject;
```

Comment:
In testing your scene DragComplete was not firing, seems this line should be 
```
targetRef.Value = targetCollider.gameObject;
```

Worth double checking the other scripts that use targets too

Reply:
There is no Collider2DVariable defined yet so I was using GameObjectVariable instead.

Reply:
I have fixed the wrong reference. It was a special case in drag completed since the event does not store the target collider. So I had to get it from the OnDragEntered and store it in the OnDragCompleted Class.  

The upgrade of fields does seem to work when I opened the scene in editor, does it not work in runtime? (If I open a scene from the old system in editor, it doesn't throw any errors.)

And why don't you like the solution you proposed? and secondly, I'm new to Unity so I don't have an alternate solution.